src/test/java/roomescape/application/ReservationServiceTest.java

```diff
@@ -0,0 +1,110 @@
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ActiveProfiles;
+import roomescape.TestRepositoryHelper;
+import roomescape.domain.reservation.ReservationSearchFilter;
+import roomescape.domain.theme.Theme;
+import roomescape.domain.timeslot.TimeSlot;
+import roomescape.domain.user.User;
+import roomescape.exception.AlreadyExistedException;
+import roomescape.exception.BusinessRuleViolationException;
+
+@DataJpaTest
+@ActiveProfiles("test")
+@Import({ReservationService.class, TestRepositoryHelper.class})
+class ReservationServiceTest {
+
+    private static final ReservationSearchFilter NONE_FILTERING = new ReservationSearchFilter(null, null, null, null);
+
+    @Autowired
+    private ReservationService service;
+    @Autowired
+    private TestRepositoryHelper repositoryHelper;
+
+    private User user;
+    private TimeSlot timeSlot;
+    private Theme theme;
+
+    @BeforeEach
+    void setUp() {
+        user = repositoryHelper.saveAnyUser();
+        timeSlot = repositoryHelper.saveAnyTimeSlot();
+        theme = repositoryHelper.saveAnyTheme();
+    }
+
+    @Test
+    @DisplayName("예약을 추가할 수 있다.")
+    void reserve() {
+        var reserved = service.reserve(user.id(), tomorrow(), timeSlot.id(), theme.id());
+
+        var reservations = service.findAllReservations(NONE_FILTERING);
+        assertThat(reservations).contains(reserved);
+    }
+
+    @Test
+    @DisplayName("예약을 삭제할 수 있다.")
+    void deleteReservation() {
+        // given
+        var reserved = service.reserve(user.id(), tomorrow(), timeSlot.id(), theme.id());
+
+        // when
+        service.removeById(reserved.id());
+
+        // then
+        var reservations = service.findAllReservations(NONE_FILTERING);
+        assertThat(reservations).doesNotContain(reserved);
+    }
+
+    @Test
+    @DisplayName("검색 필터로 예약을 조회할 수 있다.")
+    void findAllReservationsWithFilter() {
+        // given
+        var afterOneDay = service.reserve(user.id(), tomorrow(), timeSlot.id(), theme.id());
+        var afterTwoDay = service.reserve(user.id(), afterNDay(2), timeSlot.id(), theme.id());
+        var afterThreeDay = service.reserve(user.id(), afterNDay(3), timeSlot.id(), theme.id());
+
+        // when
+        var fromYesterday_toToday = new ReservationSearchFilter(theme.id(), user.id(), yesterday(), today());
+        var fromToday_toTomorrow = new ReservationSearchFilter(theme.id(), user.id(), today(), tomorrow());
+        var fromTomorrow_toThreeDays = new ReservationSearchFilter(theme.id(), user.id(), tomorrow(), afterThreeDay.dateTime().date());
+
+        assertAll(
+                () -> assertThat(service.findAllReservations(fromYesterday_toToday)).isEmpty(),
+                () -> assertThat(service.findAllReservations(fromToday_toTomorrow)).containsOnly(afterOneDay),
+                () -> assertThat(service.findAllReservations(fromTomorrow_toThreeDays)).containsExactly(afterOneDay, afterTwoDay, afterThreeDay)
+        );
```

Comment:
```suggestion
        SoftAssertions.assertSoftly(softly -> {
            softly.assertThat(service.findAllReservations(fromYesterday_toToday)).isEmpty();
            softly.assertThat(service.findAllReservations(fromToday_toTomorrow)).containsOnly(afterOneDay);
            softly.assertThat(service.findAllReservations(fromTomorrow_toThreeDays)).containsExactly(afterOneDay, afterTwoDay, afterThreeDay;
        });
```

이건 취향에 따라 봐주시면 돼요~ assertSoftly라는 친구도 있다는 것을 공유드려 보아요

Reply:
참고하겠습니다!